src/main/scala/is/hail/expr/AST.scala

```diff
@@ -426,21 +426,44 @@ case class TupleConstructor(posn: Position, elements: Array[AST]) extends BaseSt
   } yield ir.MakeTuple(irElements)
 }
 
-case class ReferenceGenomeDependentConstructor(posn: Position, fName: String, grName: String, args: Array[AST]) extends AST(posn, args) {
+case class ReferenceGenomeDependentFunctions(posn: Position, fName: String, grName: String, args: Array[AST]) extends AST(posn, args) {
   val rg = ReferenceGenome.getReference(grName)
+
   val rTyp = fName match {
     case "Variant" => rg.variantType
     case "Locus" => rg.locusType
     case "LocusInterval" => rg.intervalType
     case "LocusAlleles" => TStruct("locus" -> rg.locusType, "alleles" -> TArray(TString()))
+    case "getReferenceSequence" => TString()
     case _ => throw new UnsupportedOperationException
   }
 
-  override def typecheckThis(): Type = rTyp
+  override def typecheckThis(): Type = {
+    fName match {
+      case "getReferenceSequence" =>
+        args.map(_.`type`) match {
+          case Array(TString(_), TInt32(_), TInt32(_), TInt32(_)) =>
+          case _ => fatal(s"invalid arguments to 'getReferenceSequence'.")
```

Comment:
let's print the argument types that we did get here, so as to help the user.

Reply:
It's all checked in python, so maybe better is just make it an unchecked match